packages/amplify_datastore/ios/Classes/types/model/FlutterAuthRule.swift

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+
+import Foundation
+import Amplify
+
+public struct FlutterAuthRule {
+    
+    private let ownerField : String?
+    private let identityClaim : String?
+    private let groupClaim : String?
+    private let groups : [String]?
+    private let groupsField : String?
+    private var authStrategy : AuthStrategy!
+    private var operations : [ModelOperation]?
+    
+    init(serializedData: [String: Any]) {
+        
+        
+        self.ownerField = serializedData["ownerField"] as? String
+
+        self.identityClaim = serializedData["identityClaim"] as? String
+        
+        self.groupClaim = serializedData["groupClaim"] as? String
+        
+        self.groups = serializedData["groups"] as? [String]
+        
+        self.groupsField = serializedData["groupsField"] as? String
+        
+        self.authStrategy = stringToAuthStrategy(string: serializedData["authStrategy"] as! String)
+
+        self.operations =
+            (serializedData["operations"] as! [String]).map {
+                stringToModelOperation(string: $0)
+        }
+        
+    }
+    
+    private func stringToAuthStrategy(string: String) -> AuthStrategy{
+        switch string {
+            case "OWNER":
+                return AuthStrategy.owner
+            case "GROUPS":
+                return AuthStrategy.groups
+            case "PRIVATE":
+                return AuthStrategy.private
+            case "PUBLIC":
+                return AuthStrategy.public
+            default:
+                preconditionFailure("Could not create a AuthStrategy from \(string)")
+        }
+    }
+    
+    private func stringToModelOperation(string: String) -> ModelOperation{
+        switch string {
+            case "CREATE":
+                return ModelOperation.create
+            case "UPDATE":
+                return ModelOperation.update
+            case "DELETE":
+                return ModelOperation.delete
+            case "READ":
+                return ModelOperation.read
+            default:
+            preconditionFailure("Could not create a ModelOperation from \(string)")
+        }
+    }
+    
+    public func convertToNativeModelField() -> AuthRule{
```

Comment:
formatting and the name of the method is wrong.

Reply:
updated thank you